odl/solvers/nonsmooth/proximal_gradient_solvers.py

```diff
@@ -0,0 +1,242 @@
+        ``f.proximal``.
+    g : `Functional`
+        The function ``g`` in the problem definition. Needs to have
+        ``g.gradient``.
+    gamma : positive float
+        Step size parameter.
+    niter : non-negative int, optional
+        Number of iterations.
+
+    Other Parameters
+    ----------------
+    callback : callable, optional
+        Function called with the current iterate after each iteration.
+    lam : float or callable, optional
+        Overrelaxation step size. If callable, should take an index (zero
+        indexed) and return the corresponding step size.
+        Default: 1.0
+
+    Notes
+    -----
+    The problem of interest is
+
+    .. math::
+
+        \\min_{x \\in X} f(x) + g(x),
+
+    where the technical conditions are that
+    :math:`f : X \\rightarrow \mathbb{R}` is proper, convex and
+    lower-semicontinuous, and :math:`g : X \\rightarrow \mathbb{R}` is
+    differentiable and :math:`\\nabla g` is :math:`1 / \\beta`-Lipschitz
+    continuous.
+
+    Convergence is only guaranteed if the step length :math:`\\gamma` satisfies
+
+    .. math::
+
+       0 < \\gamma < 2 \\beta
+
+    and the parameter :math:`\\lambda` (``lam``) satisfies
+
+    .. math::
+
+       \\sum_{k=0}^\\infty \\lambda_k (\\delta - \\lambda_k) = + \\infty
+
+    where :math:`\\delta = \\min \{1, \\beta / \\gamma\}`.
+    """
+    # Starting point
+    if x not in f.domain:
+        raise TypeError('`x` {!r} is not in the domain of `f` {!r}'
+                        ''.format(x, f.domain))
+    if x not in g.domain:
+        raise TypeError('`x` {!r} is not in the domain of `g` {!r}'
+                        ''.format(x, g.domain))
+
+    # Step size parameter
+    gamma, gamma_in = float(gamma), gamma
+    if gamma <= 0:
+        raise ValueError('`gamma` must be positive, got {}'.format(gamma_in))
+
+    # Number of iterations
+    if not isinstance(niter, int) or niter < 0:
+        raise ValueError('`niter` {} not understood'
+                         ''.format(niter))
+
+    lam_in = kwargs.pop('lam', 1.0)
+    lam = lam_in if callable(lam_in) else lambda _: float(lam_in)
+
+    # Get the proximals
+    f_prox = f.proximal(gamma)
+
+    # Create temporary
+    tmp = x.space.element()
+
+    for k in range(niter):
+        lam_k = lam(k)
+
+        # x - tau grad_g (x)
+        tmp.lincomb(1, x, -gamma, g.gradient(x))
+
+        # Update x
+        x.lincomb(1 - lam_k, x, lam_k, f_prox(tmp))
+
+        if callback is not None:
+            callback(x)
+
+
+def accelerated_proximal_gradient(x, f, g, gamma, niter, callback=None,
+                                  **kwargs):
+    """Accelerated proximal gradient algorithm for convex optimization.
+
+    The method is known as "Fast Iterative Soft-Thresholding Algorithm"
+    (FISTA).
+
+    Solves the convex optimization problem::
+
+        min_{x in X} f(x) + g(x)
+
+    where ``g`` is differentiable and the proximal operator of ``f`` is known.
```

Comment:
Can you point out the difference to the non-accelerated version in one sentence? Like writing out the update rule for `t`, and what it is.